contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal constant umbra = IUmbra(0xFb2dc580Eed955B528407b4d36FfaFe3da685401);
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  struct SendToken {
+    address receiver;
+    address tokenAddr;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+  
+  error ValueMismatch();
+  event Log(address indexed caller, uint256 indexed value, string message);
+
+  function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) external payable {
+    uint256 valAccumulator;
```

Comment:
nit: would prefer a more descriptive name like `totalValueSending` or something like this

Reply:
What about `valueSentAccumulator`? I do like `accumulator` as it's a common descriptor for variables like this